lib/fast-led-light-controller/FastLedLightController.cpp

```diff
@@ -1,5 +1,21 @@
 #include "FastLedLightController.hpp"
 
+#include <cstdio>
+
+int print_compiler_version() {
+  std::printf("====================");
+#if defined __clang_major__
+  std::printf("clang version %d.%d\n", __clang_major__, __clang_minor__);
+#elif defined __GNUC__
+  std::printf("gcc version %d.%d\n", __GNUC__, __GNUC_MINOR__);
+#else
+  std::printf("Unknown compiler\n");
+#endif
+  std::printf("====================");
+}
+
+static int dummy = print_compiler_version();
```

Comment:
Why the dummy here? So you can control if it's included or not?

Reply:
I wanted to run code regardless of the "entry point" of the program. The easiest way to do that is to create a global variable where constructing it has the side effect I need. I just changed to a version that just prints it at compile time by creating a template error message, so that I can find the output easier.